cmd/gateway-main.go

```diff
@@ -185,11 +185,24 @@ func gatewayMain(ctx *cli.Context) {
 	}
 
 	// First argument is selected backend type.
-	backendType := ctx.Args().First()
+	backendType := ctx.Args().Get(0)
+	// Second argument is the endpoint address (optional)
+	endpointAddr := ctx.Args().Get(1)
+	// Third argument is the address flag
+	serverAddr := ctx.String("address")
```

Comment:
> Additionally not sure why gateway is not failing to return error on port already in use.

We don't have two codes that are trying to allocate a port, but actually, the gateway keeps callting itself infinitely.



> Here valid value format of endpointAddr is "",

empty endpoint addr is not valid anymore. We don't need this case. I updated the PR.



> If the PR scope is only to last point only, I am not sure how critical to fix that case.

It is little annoying to not address this use case because:
- the old shown error is non-understandable to our users
`2017/05/09 10:53:09 http: Accept error: accept tcp [::]:9000: accept4: too many open files; retrying in 5ms`
- this use case allocates all available resources which could have bad effect on the system.